src/Engine/ProtoCore/FFI/CLRObjectMarshler.cs

```diff
@@ -494,24 +494,21 @@ private ICollection ToICollection(StackValue dsObject, ProtoCore.Runtime.Context
     /// <summary>
     /// Marshales string as array of chars
     /// </summary>
-    class StringMarshaler : CollectionMarshaler
+    class StringMarshaler : PrimitiveMarshler 
```

Comment:
We need to review the implementation of CLRObjectMarshaler class, which dispatches call to one of these marshalers, because string in C# is represented as collection. Just make sure that we are able to marshal strings properly.